sql/mysqld.cc

```diff
@@ -4421,6 +4424,83 @@ a file name for --log-bin-index option", opt_binlog_index_name);
     unireg_abort(MYSQLD_ABORT_EXIT);
   }
 
+  if (opt_libcoredumper)
+  {
+#ifndef HAVE_LIBCOREDUMPER
+    sql_print_warning(
+        "This version of MySQL has not been compiled with "
+        "libcoredumper support, ignoring --coredumper argument");
+#else
+    if (opt_corefile)
+    {
+      sql_print_warning(
+          "Started with --core-file and --coredumper. "
+          "--coredumper will take precedence.");
+    }
+    if (opt_libcoredumper_path != NULL)
+    {
+      /* validate path */
+      if (!is_valid_log_name(opt_libcoredumper_path,
+                             strlen(opt_libcoredumper_path)))
+      {  //filename contain .cnf or .ini on it
+        sql_print_error("Variable --coredumper cannot be set to value %s",
+                        opt_libcoredumper_path);
+        unireg_abort(MYSQLD_ABORT_EXIT);
+      }
+      char   libcoredumper_dir[FN_REFLEN];
+      size_t libcoredumper_dir_length;
+      size_t opt_libcoredumper_path_length= strlen(opt_libcoredumper_path);
+      (void)dirname_part(libcoredumper_dir, opt_libcoredumper_path,
+                         &libcoredumper_dir_length);
+
+      if (!libcoredumper_dir_length)
+      {
+        sql_print_error("Error processing --coredumper path: %s",
+                        opt_libcoredumper_path);
+        unireg_abort(MYSQLD_ABORT_EXIT);
+      }
+      size_t libcoredumper_file_length=
+          opt_libcoredumper_path_length - libcoredumper_dir_length;
+      if (libcoredumper_file_length == 0)  //path is a directory
+      {
+        libcoredumper_file_length= 19;  //file is set to core.yyyymmddhhmmss
+      }
+      else
+      {
+        libcoredumper_file_length+= 17;  //file gets .yyyymmddhhmmss appended
+      }
+      if (opt_libcoredumper_path_length > FN_REFLEN)
+      {  // path is too long
+        sql_print_error("Variable --coredumper set to a too long path");
+        unireg_abort(MYSQLD_ABORT_EXIT);
+      }
+      if (libcoredumper_file_length > FN_LEN)
+      {  // filename is too long
+        sql_print_error("Variable --coredumper set to a too long filename");
+        unireg_abort(MYSQLD_ABORT_EXIT);
+      }
+      if (my_access(libcoredumper_dir, F_OK))
+      {
+        sql_print_error(
```

Comment:
What if it gets removed while the server is running?

Reply:
Done, I now validate the path again, in case of any error we revert back to have coredump written on datadir